src/python/pants/util/rwbuf.py

```
@@ -79,3 +79,29 @@ def __init__(self, backing_file):
 
   def do_write(self, s):
     self._io.write(s)
+
+
+class StringWriter(object):
+  """A write-only buffer which accepts strings and writes to another buffer which accepts bytes.
+
+  Writes strings as utf-8.
+
+  This is write-only because it's unclear whether seeking should seek by code-point or byte, and
+  implementing the former is non-trivial. If you need to read, read from the underlying buffer's
+  bytes.
+  """
+
+  def __init__(self, underlying):
+    # Called buffer to mirror how sys.stdout and sys.stderr expose this.
+    self.buffer = underlying
+
+  def write(self, s):
+    if not isinstance(s, str):
+      raise ValueError('Expected unicode str, not {}, for argument {}'.format(type(s), s))
```

Great, failing fast is very helpful. Thanks for this.